vault/quotas/quotas.go

```diff
@@ -377,8 +387,8 @@ func (m *Manager) quotaNamesLocked(qType Type) ([]string, error) {
 
 // QuotaByID queries for a quota rule in the db for a given quota ID
 func (m *Manager) QuotaByID(qType string, id string) (Quota, error) {
-	m.lock.RLock()
-	defer m.lock.RUnlock()
+	m.quotaLock.RLock()
```

Comment:
If we're going to drop the lock from QueryQuota, I see no reason why it should stay for these other equivalent QuotaBy methods.

Reply:
Thanks, good point. I've cleaned up everywhere locks are used, and it looks good now.